src/pages/ServiceDetails/ServiceMetrics.tsx

```diff
@@ -211,6 +130,155 @@ class ServiceMetrics extends React.Component<ServiceId, ServiceMetricsState> {
     // 2 decimal digits
     return Math.round(val * 100) / 100;
   }
+
+  render() {
+    return (
+      <div>
+        <MetricsOptionsBar onOptionsChanged={this.onOptionsChanged} />
+        {this.renderMetrics()}
+      </div>
+    );
+  }
+
+  renderMetrics() {
+    if (this.state.loading) {
+      return <Spinner loading={true} />;
+    }
+    return (
+      <div className="card-pf">
+        <div className="row row-cards-pf">
+          <div className="col-xs-12">
+            <div className="card-pf-accented card-pf-aggregate-status">
+              <div className="card-pf-title">
+                <span className="fa fa-heart" />
+                Health
+                {this.renderHealth()}
+              </div>
+            </div>
+          </div>
+        </div>
+        <div className="row row-cards-pf">
+          <div className="col-xs-6">
+            <div className="card-pf-accented card-pf-aggregate-status">
+              <h3 className="card-pf-title">
+                <span className="fa fa-bar-chart" />
+                Input
+              </h3>
+              <div className="card-pf-body">
+                {this.renderMetric('requestCountIn', 'Request count rate', this.state.requestCountIn)}
```

Comment:
My first thought is that renderMetric and renderHistogram perhaps might have their own component, just to reduce the complexity of this render.
But it is fine, just a comment for future refactorings (when it makes sense ;-).